### PR TITLE
Refactor SYNC-3840 [v119] Remove call to SQLCipher migration

### PR DESCRIPTION
**Pull request by skhamis**
Removes the calls to SQLCipher migration. We should do this first before actually removing it from a-s to prevent any breaking changes needed.

a-s tracking issue: https://github.com/mozilla/application-services/issues/5777
a-s tracking epic: https://mozilla-hub.atlassian.net/browse/SYNC-3838

## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/SYNC-3840)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/16055)

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: Providers/Profile.swift
Comment (nbhasin2) [inline Providers/Profile.swift:623]: Would be good to add some error handling if possible?
I see it it's inside lazy var but want to make sure we can avoid force try!

```
//EXAMPLE
let directory: String
do {
    directory = try files.getAndEnsureDirectory()
} catch {
    // error handling
    return
}

let sqlCipherDatabasePath = URL(fileURLWithPath: directory, isDirectory: true).appendingPathComponent("logins.db").path
```

File: Storage/Rust/RustLogins.swift
Comment (skhamis) [reply Storage/Rust/RustLogins.swift:813]: This was actually more in general if this was the best way to clear it and I think it is so I'll go ahead and remove the comment!

File: Providers/Profile.swift
Comment (skhamis) [reply Providers/Profile.swift:623]: (as discussed) unfortunately going this route ran into a bit of an issue due to the need to return `RustLogins` and given it's a `lazy var` we can't throw here easily.

**Review by nbhasin2 — APPROVED** [review]
R+ except for the nits and we discussed to revisit the force unwrap
@skhamis could I request you add a `TODO: #TicketNumber` so we know to revisit 🙏🏼

**Comment by nbhasin2** [comment]
@skhamis when you say we should not land this, do you mean not land it on `main` for v118 or not ship it?

**Comment by skhamis** [comment]
> @skhamis when you say we should not land this, do you mean not land it on `main` for v118 or not ship it?

Since this is a change that will affect users (current telemetry about <1% of total users), I believe product would like to coordinate a notice of some sort to communicate to users that they _may_ lose data if upgrading from < v40 to this newer version. I'm fine with landing after we get the OK from that! @dnarcese

**Comment by nbhasin2** [comment]
Adding a comment so it doesn't get marked as stale cc @skhamis

**Comment by skhamis** [comment]
> Adding a comment so it doesn't get marked as stale cc @skhamis

Thanks @nbhasin2! Is it possible to get Bitrise to run on this? I believe we're good to land this!

**Pull request merged**
